src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs

```diff
@@ -39,15 +39,24 @@ public virtual async Task<long> GetTotalChangesCountAsync(DateTime? startDate, D
             }
             else
             {
+                // Get added and modified products count
+                using (var repository = _catalogRepositoryFactory())
+                {
+                    result = await repository.Items.CountAsync(i => i.ParentId == null && ((i.ModifiedDate >= startDate && i.ModifiedDate <= endDate) || (i.CreatedDate >= startDate && i.CreatedDate <= endDate)));
```

Comment:
IS this query working as expected if startDate or endDate = null?

Reply:
It is an excepted case. In that case, the system should throw an argument exception.  It was done.

Reply:
The same validation should be done in GetTotalChangesCountAsync, isn't it?